### ISSUE TITLE
An exception was thrown in the copyToRealm method in the proxy class  io.realm.PSTripRealmProxy: Annotation processor may not have been executed.

### ISSUE THREAD

**Issue by alin-rosu** [realm/realm-java#1186]
I have the following code that keeps giving me an error like this:
"An exception was thrown in the copyToRealm method in the proxy class  io.realm.PSTripRealmProxy: Annotation processor may not have been executed."

This is the code:
public void addToList(PSTrip psTrip){
        try{
            PSTrip insideAlready = getListOfCompletedTripsByID(psTrip.getId());
            Log.i("","continue init inside already:" + insideAlready);
            if(insideAlready == null){
                Log.i("","continue init will copy to realm");
                realm.beginTransaction();
                realm.copyToRealm(psTrip);
                realm.commitTransaction();
                Log.i("","continue init will copy to realm finished");
            }else{
                Log.i("","continue init will copyorupdate to realm");
                realm.beginTransaction();
                realm.copyToRealmOrUpdate(psTrip);
                realm.commitTransaction();
                Log.i("","continue init will copyorupdate to realm finished");
            }
        }catch (Exception e){
            Log.i("","continue init error trying to add to realm" + e.getMessage());
            realm.cancelTransaction();
        }
    }

This is how my PSTrip class looks like:
public class PSTrip extends RealmObject {
    private boolean valid;
    private String type;
    private String travel_mode;
    private String id;
    private Owner_data owner_data;
    private int distance;
    private RealmObj errors;
    private String name;
    private double checkinLat;
    private double checkinLng;
    private double checkoutLng;
    private double checkoutLat;
    private String icon;
    private String status;
    private Destination destination;
    private int checkout_time;
    private int checkin_time;
    private Route route;
    private String owner;
    private String vehicle;
    private Flight flight;
    @SerializedName("last_updated")
    private int lastUpdated;
    @SerializedName("steps")
    private RealmList<TripStep> tripSteps = new RealmList<TripStep>();
    @SerializedName("planned_route")
    private Planned_Route plannedRoute;
    private Group group;
    private boolean isRoaming = false;

    public boolean getIsRoaming() {
        return isRoaming;
    }

    public void setIsRoaming(boolean isRoaming) {
        this.isRoaming = isRoaming;
    }

    public Group getGroup() {
        return group;
    }

    public void setGroup(Group group) {
        this.group = group;
    }

    public int getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(int lastUpdated) {
        this.lastUpdated = lastUpdated;
    }

    public RealmList<TripStep> getTripSteps() {
        return tripSteps;
    }

    public void setTripSteps(RealmList<TripStep> steps) {
        this.tripSteps = steps;
    }

    public String getVehicle() {
        return vehicle;
    }

    public void setVehicle(String vehicle) {
        this.vehicle = vehicle;
    }

    public Flight getFlight() {
        return flight;
    }

    public void setFlight(Flight flight) {
        this.flight = flight;
    }

    public boolean getValid() {
        return valid;
    }

    public void setValid(boolean valid) {
        this.valid = valid;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getTravel_mode() {
        return travel_mode;
    }

    public void setTravel_mode(String travel_mode) {
        this.travel_mode = travel_mode;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Owner_data getOwner_data() {
        return owner_data;
    }

    public void setOwner_data(Owner_data owner_data) {
        this.owner_data = owner_data;
    }

    public int getDistance() {
        return distance;
    }

    public void setDistance(int distance) {
        this.distance = distance;
    }

    public RealmObj getErrors() {
        return errors;
    }

    public void setErrors(RealmObj errors) {
        this.errors = errors;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Destination getDestination() {
        return destination;
    }

    public void setDestination(Destination destination) {
        this.destination = destination;
    }

    public int getCheckout_time() {
        return checkout_time;
    }

    public void setCheckout_time(int checkout_time) {
        this.checkout_time = checkout_time;
    }

    public int getCheckin_time() {
        return checkin_time;
    }

    public void setCheckin_time(int checkin_time) {
        this.checkin_time = checkin_time;
    }

    public Route getRoute() {
        return route;
    }

    public void setRoute(Route route) {
        this.route = route;
    }

    public String getOwner() {
        return owner;
    }

    public void setOwner(String owner) {
        this.owner = owner;
    }

    public PSTrip() {
    }

    public PSTrip(String id) {
        this.id = id;
    }

    public double getCheckoutLng() {
        return checkoutLng;
    }

    public void setCheckoutLng(double checkoutLng) {
        this.checkoutLng = checkoutLng;
    }

    @Override
    public boolean isValid() {
        return valid;
    }

    public double getCheckinLat() {
        return checkinLat;
    }

    public void setCheckinLat(double checkinLat) {
        this.checkinLat = checkinLat;
    }

    public double getCheckinLng() {
        return checkinLng;
    }

    public void setCheckinLng(double checkinLng) {
        this.checkinLng = checkinLng;
    }

    public double getCheckoutLat() {
        return checkoutLat;
    }

    public void setCheckoutLat(double checkoutLat) {
        this.checkoutLat = checkoutLat;
    }

    public boolean isRoaming() {
        return isRoaming;
    }

    public void setRoaming(boolean isRoaming) {
        this.isRoaming = isRoaming;
    }

    public Planned_Route getPlannedRoute() {
        return plannedRoute;
    }

    public void setPlannedRoute(Planned_Route plannedRoute) {
        this.plannedRoute = plannedRoute;
    }
}

Any ideea why this is happening?

**Comment by PavelSynek** [comment]
"Annotation processor may not have been executed" means that there was a problem parsing your model class (would be great to have better error text).

You probably shouldn't override [`isValid`](https://realm.io/docs/java/latest/api/io/realm/RealmObject.html#isValid--).

What is private `private RealmObj errors;`?

What is `private RealmList tripSteps = new RealmList();`? According to [docs](https://realm.io/docs/java/latest/#why-can-i-not-create-a-realmlist-myself), you can't create RealmList

**Comment by kneth** [comment]
As @PavelSynek has answered and no reply, I'm closing the issue. @alin-rosu If you're still having problems with this, please open an issue.

**Comment by alin-rosu** [comment]
After playing a bit with it, I found the issue. I had on a class set a
primary key with "@PrimaryKey and I was trying to do
"realm.copyOrUpdateToRealm" which is where my code broke and started to
give me the "Annotation processor" error. I took out the primary key, cause
it wasn't really necessary, and I now use "copyToRealm" instead of
copyOrUpdateToRealm. Also made a function to check before adding, and it
the item exists, then it will remove it before adding the new one, this way
avoiding the issue :)
